### ISSUE TITLE
I cannot reach the ori accuracy by training the ResNeXt-50 on the ImageNet. 

### ISSUE THREAD

**Issue by PoonKinWang** [pytorch/vision#3064]
I use the ['PyTorch ImageNet Training' example](https://github.com/pytorch/examples/tree/master/imagenet) and the ['models'](https://github.com/pytorch/vision/tree/master/torchvision/models) of TorchVision 0.4.2 to train ResNeXt-50 twice but got 23.52% and 23.57% (Top-1) on ImageNet Val set, which do not reach the ori Err. (22.2%). Besides, I find that the hyper parameters setting of ['PyTorch ImageNet Training' example](https://github.com/pytorch/examples/tree/master/imagenet) is same to the [original paper](https://arxiv.org/abs/1611.05431). Can you give me some advices for training to reach the ori Err. ?

The val acc alongside training is shown below: 
<summary>
logs
<details>
Top-1
16.894
32.488
36.116
40.272
45.394
46.328
50.126
50.336
52.242
54.414
53.096
54.438
55.662
54.972
55.902
57.204
54.932
57.068
55.586
56.9
58.018
56.67
58.564
57.272
58.224
57.736
57.816
58.292
57.618
56.664
70.7
71.502
72.04
72.452
72.69
72.754
73.03
72.996
72.504
72.812
72.318
72.294
72.584
72.318
72.42
72.528
72.238
72.14
71.76
71.91
72.282
72.508
72.156
71.424
72.3
72.48
72.42
72.61
72.61
72.178
75.62
75.86
76.16
76.184
76.26
76.252
76.376
76.3
76.404
76.48
76.326
76.368
76.304
76.386
76.462
76.36
76.452
76.396
76.258
76.308
76.334
76.228
76.252
76.304
76.15
76.298
76.362
76.15
76.17
76.058
</details>
</summary>

EDIT: (vfdev-5) I updated the message and put the training logs into summary/details block.

**Comment by vfdev-5** [comment]
@PoonKinWang it is difficult to say for me where is the issue. Please, check reference script here for the classification task:
https://github.com/pytorch/vision/tree/master/references/classification
Maybe, using them you can obtain desired accuracy.

**Comment by PoonKinWang** [comment]
> @PoonKinWang it is difficult to say for me where is the issue. Please, check reference script here for the classification task:
> https://github.com/pytorch/vision/tree/master/references/classification
> Maybe, using them you can obtain desired accuracy.

Thanks for your reply. 
The setting of ['PyTorch ImageNet Training' example](https://github.com/pytorch/examples/tree/master/imagenet) is same to what you mentioned, and both of them were the official code of TorchVision. Thus, I think the code you mentioned cannot work as well. Could you offer any training logs of ResNeXt-50?

**Comment by vfdev-5** [comment]
> Could you offer any training logs of ResNeXt-50?

Unfortunately, I do not have these logs, maybe, @fmassa still has them.

**Comment by PoonKinWang** [comment]
![image](https://user-images.githubusercontent.com/55362783/100661988-42dc3600-338f-11eb-989c-e31829c9e226.png)

**Comment by PoonKinWang** [comment]
> > Could you offer any training logs of ResNeXt-50?
> 
> Unfortunately, I do not have these logs, maybe, @fmassa still has them.
> 
> @PoonKinWang By the way, do you train the model on the same infrastructure setup 8x V100 GPUs ?
> and what are the software versions you use ?
> 
> ```
> pip list | grep torch
> ```

![Uploading image.png…]()

**Comment by vfdev-5** [comment]
> I trained on two 2080ti.

@PoonKinWang if your total batch size is not 256 (=8 * 32) as for 8 GPUs, then you have to rescale the learning rate like 
```
total_batch_size = ...
lr = 0.1 * (total_batch_size / 256.0)
```
(as ref: https://arxiv.org/abs/1706.02677)

**Comment by PoonKinWang** [comment]
> > I trained on two 2080ti.
> 
> @PoonKinWang if your total batch size is not 256 (=8 * 32) as for 8 GPUs, then most probably you have to rescale the learning rate like
> 
> ```
> total_batch_size = ...
> lr = 0.1 * (total_batch_size / 256.0)
> ```
> 
> (as ref: https://arxiv.org/abs/1706.02677)

On that two training, I used total batch size of 256 on two GPUs.

**Comment by vfdev-5** [comment]
@PoonKinWang do you DistributedDataParallel (not DataParalle) right ?  (https://pytorch.org/tutorials/intermediate/ddp_tutorial.html#comparison-between-dataparallel-and-distributeddataparallel)

**Comment by PoonKinWang** [comment]
> @PoonKinWang do you mean DistributedDataParallel (not DataParalle) right ? (https://pytorch.org/tutorials/intermediate/ddp_tutorial.html#comparison-between-dataparallel-and-distributeddataparallel)

I mean DataParalle.

**Comment by PoonKinWang** [comment]
> @PoonKinWang do you mean DistributedDataParallel (not DataParallel) right ? (https://pytorch.org/tutorials/intermediate/ddp_tutorial.html#comparison-between-dataparallel-and-distributeddataparallel)

I use the train.py as you mentioned to train again but still fail to reach the Acc. as the ori paper provide (22.2%). I get 22.52%, 22.67% and 22.81% as three independent training results.

**Comment by PoonKinWang** [comment]
> @PoonKinWang it is difficult to say for me where is the issue. Please, check reference script here for the classification task:
> https://github.com/pytorch/vision/tree/master/references/classification
> Maybe, using them you can obtain desired accuracy.

I use the train.py as you mentioned to train again but still fail to reach the Acc. as the ori paper provide (22.2%). I get 22.52%, 22.67% and 22.81% as three independent training results.

**Comment by PoonKinWang** [comment]
@vfdev-5 I use the train.py as you mentioned to train again but still fail to reach the Acc. as the ori paper provide (22.2%). I get 22.52%, 22.67% and 22.81% as three independent training results.

**Comment by fmassa** [comment]
Hi,

Here is the training log for ResNeXt50 32x4d for the model that we uploaded for torchvision https://gist.github.com/fmassa/aa7f0b0890317cddceeec6fa29fd221a

It reaches best accuracies of
```
Acc@1 77.618 Acc@5 93.698
```
in epoch 97.

We used the training scripts present in `references`, without sync-batchnorm, on 8 GPUs each with batch size of 32, trained for 100 epochs (as described in https://github.com/pytorch/vision/tree/master/references/classification#resnext-50-32x4d).
This means that in your setup the batch size per GPU should be 128, which will have different statistics for batch norm and could explain the difference in results.

I believe I've answered your question and as such I'm closing this issue, but let us know if you have further questions
